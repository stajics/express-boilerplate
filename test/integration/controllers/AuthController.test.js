require('../../index');

describe(':signup', () => {
  jestTest('[POST /auth/signup] Should signup', async (done) => {
    // client.setToken(fixtures.users[0].token);
    const res = await client.post('/auth/signup', {
      email: 'test1@test.com',
      password: 'password',
      firstName: 'name',
      lastName: 'name',
    });
    delete res.body.data.token; // remove changing attribute
    delete res.body.data._id; // remove changing attribute
    // expect({ expected: 'expected' }).toMatchSnapshot(); run this first for ttd
    expect(res.body).toMatchSnapshot();
    done();
  });

  jestTest('[POST /auth/signup] Should fail signup (no password)', async (done) => {
    const res = await client.post('/auth/signup', {
      email: 'test1@test.com',
      firstName: 'name',
      lastName: 'name',
    });
    expect(res.body).toMatchSnapshot();
    done();
  });
});

describe(':login', () => {
  jestTest('[POST /auth/login] Should login', async (done) => {
    const res = await client.post('/auth/login', {
      email: fixtures.users[0].email,
      password: fixtures.users[0].password,
    });
    delete res.body.data.token;
    expect(res.body).toMatchSnapshot();
    done();
  });

  jestTest('[POST /auth/login] Should fail login (bad email)', async (done) => {
    const res = await client.post('/auth/login', {
      email: 'bad@email.com',
      password: fixtures.users[0].password,
    });
    expect(res.body).toMatchSnapshot();
    done();
  });

  jestTest('[POST /auth/login] Should fail login (bad password)', async (done) => {
    const res = await client.post('/auth/login', {
      email: fixtures.users[0].email,
      password: 'bad_password',
    });
    expect(res.body).toMatchSnapshot();
    done();
  });

  jestTest('[POST /auth/login] Should fail login (no email)', async (done) => {
    const res = await client.post('/auth/login', {
      password: fixtures.users[0].password,
    });
    expect(res.body).toMatchSnapshot();
    done();
  });
});
