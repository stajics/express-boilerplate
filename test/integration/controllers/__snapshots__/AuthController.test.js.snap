// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`:login [POST /auth/login] Should fail login (bad email) 1`] = `
Object {
  "error": Object {
    "code": 1,
    "message": "Bad credentials",
  },
}
`;

exports[`:login [POST /auth/login] Should fail login (bad password) 1`] = `
Object {
  "error": Object {
    "code": 1,
    "message": "Bad credentials",
  },
}
`;

exports[`:login [POST /auth/login] Should fail login (no email) 1`] = `
Object {
  "error": Object {
    "errors": Array [
      Object {
        "field": Array [
          "email",
        ],
        "location": "body",
        "messages": Array [
          "\\"email\\" is required",
        ],
        "types": Array [
          "any.required",
        ],
      },
    ],
    "status": 400,
    "statusText": "Payload Validation Error",
  },
}
`;

exports[`:login [POST /auth/login] Should login 1`] = `
Object {
  "data": Object {
    "__v": 0,
    "_id": "189b5b4aa79dff2f7fe6659d",
    "email": "user1@test.com",
    "firstName": "user1_name",
    "lastName": "user1_lastname",
  },
}
`;

exports[`:signup [POST /auth/signup] Should fail signup (no password) 1`] = `
Object {
  "error": Object {
    "errors": Array [
      Object {
        "field": Array [
          "password",
        ],
        "location": "body",
        "messages": Array [
          "\\"password\\" is required",
        ],
        "types": Array [
          "any.required",
        ],
      },
    ],
    "status": 400,
    "statusText": "Payload Validation Error",
  },
}
`;

exports[`:signup [POST /auth/signup] Should signup 1`] = `
Object {
  "data": Object {
    "__v": 0,
    "email": "test1@test.com",
    "firstName": "name",
    "lastName": "name",
  },
}
`;
